pkg/webhook/pod/mutating/sidecarset.go

```diff
@@ -210,7 +210,8 @@ func (h *PodCreateHandler) getSpecificRevisionSidecarSetForPod(sidecarSet *appsv
 	var err error
 	var matchedSidecarSet *appsv1alpha1.SidecarSet
 	for _, revision := range revisions {
-		if sidecarcontrol.GetPodSidecarSetControllerRevision(sidecarSet.Name, pod) == revision.Name {
+		if sidecarcontrol.GetPodSidecarSetControllerRevision(sidecarSet.Name, pod) == revision.Name ||
+			sidecarcontrol.GetPodSidecarSetRevision(sidecarSet.Name, pod) == sidecarcontrol.GetSidecarSetRevision(sidecarSet) { // backward compatible
```

Comment:
plz add comments for SidecarSetUpgradeSpec struct, and explain difference between SidecarSetHash and SidecarSetControllerRevision